packages/core/types/src/notification/mutations.ts

```diff
@@ -21,6 +21,10 @@ export interface CreateNotificationDTO {
    * The data that gets passed over to the provider for rendering the notification.
    */
   data?: Record<string, unknown> | null
+  /**
+   * The metadata that gets passed over to the provider.
+   */
+  providerContext?: Record<string, unknown> | null
```

Comment:
should be snake_cased

```suggestion
  provider_context?: Record<string, unknown> | null
```